src/openrct2/ride/Ride.h

```diff
@@ -746,6 +747,18 @@ enum
     BREAKDOWN_COUNT
 };
 
+#ifdef ENABLE_SCRIPTING
+static const OpenRCT2::Scripting::DukEnumMap<uint8_t> BreakdownMap // The types of breakdowns.
```

Comment:
Not a fan of including all of Duktape.hpp in Ride.h. This is quite a heavy header! As Ride.h is included in many places, this will likely increase compilation times quite a bit.

NB: you can use EnumMap instead of DukEnumMap; the two are compatible.

Please move this map out of Ride.h, though. It can be defined in a separate header, only included in Ride.cpp and ScRide.cpp.